test/functional/change_stream_spec.test.js

```diff
@@ -37,11 +37,13 @@ describe('Change Stream Spec', function() {
         const sDB = suite.database_name;
         const sColl = suite.collection_name;
         const configuration = this.configuration;
-        return Promise.all(ALL_DBS.map(db => gc.db(db).dropDatabase({ w: 'majority' })))
-          .then(() => gc.db(sDB).createCollection(sColl))
-          .then(() => gc.db(suite.database2_name).createCollection(suite.collection2_name))
-          .then(() => configuration.newClient({}, { monitorCommands: true }).connect())
-          .then(client => {
+        const steps = [() => gc.db(sDB).createCollection(sColl)];
```

Comment:
am I reading this correctly that the refactor here was because a suite might not specify a `database2_name`? You can also achieve the same result by conditionally executing code in a `then` block:
```
.then(() => {
  if (suite.database2_name && suite.collection2_name) {
    return gc.db(suite.database2_name).createCollection(suite.collection2_name);
  }
})
```

I think it makes it somewhat more readable
})

Reply:
Yes and agreed, I'll clean that up.